twine-upload.sh

```diff
@@ -48,3 +48,29 @@ TWINE_USERNAME="$INPUT_USER" \
 TWINE_PASSWORD="$INPUT_PASSWORD" \
 TWINE_REPOSITORY_URL="$INPUT_REPOSITORY_URL" \
   exec twine upload ${TWINE_EXTRA_ARGS} ${INPUT_PACKAGES_DIR%%/}/*
+
+if [[ ${INPUT_PRINT_HASH,,} != "false" ]] ; then
```

Comment:
This will not work because `exec` is a syscall that replaces the current process. Basically, it'll interrupt the script and run `twine` instead. You'll need to move the code above. Also, maybe add "or verbose" to the condition.
Oh, and please move the Python snippet to a standalone module instead of embedding it into the shell script, from there, it'll be easier to review it separately (it'd be easier to notice the need to use context managers, for example).


P.S. You can test this PR if you point some repo to your fork + specify this PR branch.

Reply:
@webknjaz Thanks for your review!

Is it possible to use `${INPUT_PACKAGES_DIR%%/}` in Python if it is moved out from the shell script?